js/src/forum/states/SearchState.js

```diff
@@ -0,0 +1,73 @@
+export default class SearchState {
+  constructor(cachedSearches = [], searchRoute = 'index') {
+    this.searchRoute = searchRoute;
+
+    this.cachedSearches = cachedSearches;
+  }
+
+  /**
+   * Get URL parameters that stick between filter changes.
+   *
+   * @return {Object}
+   */
+  stickyParams() {
+    return {
+      sort: m.route.param('sort'),
+      q: m.route.param('q'),
+    };
+  }
+
+  /**
+   * Get parameters to pass to the DiscussionList component.
+   *
+   * @return {Object}
+   */
+  params() {
+    const params = this.stickyParams();
+
+    params.filter = m.route.param('filter');
+
+    return params;
+  }
+
+  /**
+   * Return the current search query, if any. This is implemented to activate
+   * the search box in the header.
+   *
+   * @see Search
+   * @return {String}
+   */
+  searching() {
+    return this.params().q;
+  }
+
+  /**
+   * Redirect to the index page using the given sort parameter.
+   *
+   * @param {String} sort
+   */
+  changeSort(sort) {
+    const params = this.params();
+
+    if (sort === Object.keys(app.cache.discussionList.sortMap())[0]) {
```

Comment:
I suppose this will need changing after #2150 is merged?

Reply:
Yep!

Reply:
To play devil's advocate, how do we track these cross-dependencies?

Reply:
We don't. Probably should, but I don't know of a way that would be too great. I suppose TODO comments in the code, or editing the description of each issue could work?

Reply:
I guess we can leave a TODO comment in the PR discussion of #2150, then merge this one here first.